tests/release-build.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+# test name and purpose
+echo ''
+echo '                        ##### Release Build Test #####'
+echo ''
+echo 'The purpose of this test is to ensure that nodeos was built without debugging'
+echo 'symbols. Debugging symbols enable software engineers to inspect and control a'
+echo 'running program with a debugging tool, but they significantly slow down'
+echo 'performance-critical applications like nodeos. Anyone intending to build and'
+echo 'install nodeos from source should perform a "release build," which excludes'
+echo 'debugging symbols to generate faster and lighter binaries.'
+echo ''
+# check for jq
+if ! $(jq --version 1>/dev/null); then
+    echo 'ERROR: Test requires jq, but jq was not found in your PATH!'
+    echo ''
+    exit 1
+fi
+# find nodeos
+[[ -z "$EOSIO_ROOT" && $(git --version) ]] && export EOSIO_ROOT="$(git rev-parse --show-toplevel)"
+[[ -z "$EOSIO_ROOT" ]] && export EOSIO_ROOT="$(echo $(pwd)/ | grep -ioe '.*/eos/' -e '.*/eosio/' -e '.*/build/' | sed 's,/build/,/,')"
+if [[ ! -f "$EOSIO_ROOT/build/bin/nodeos" && ! -f "$EOSIO_ROOT/build/programs/nodeos/nodeos" ]]; then
```

Comment:
You know that this test file needs to be located within the `tests` subdirectory of the build directory. So you can just get the path of the source file and then find nodeos relative to it (`../programs/nodeos/nodes`) rather than dealing with git commands or assuming the build folder is named `build`.